src/test/java/net/openhft/chronicle/queue/impl/single/RollCycleMultiThreadStressTest.java

```diff
@@ -477,6 +520,10 @@ private int write(ExcerptAppender appender) {
             try (DocumentContext writingDocument = appender.writingDocument()) {
                 final long documentAcquireTimestamp = System.nanoTime();
                 value = wrote.getAndIncrement();
+                if (value >= expectedNumberOfMessages) {
+                    writingDocument.rollbackOnClose();
+                    return value;
+                }
```

Comment:
I needed to do this because threads can open `DocumentContext`s concurrently when double buffering is enabled, so in those circumstances they'd sometimes write more than `expectedNumberOfMessages`

Reply:
Combine that with being potentially out of order and it becomes very hard to check on the reader side.

Reply:
maybe add a comment to that code explaining why?